backend/utils/passport.ts

```diff
@@ -0,0 +1,53 @@
+import passport from 'passport'
+import passportGitHub from 'passport-github'
+
+import config from './config'
+
+import User, { UserType } from '../types/user'
+
+const GithubStrategy = passportGitHub.Strategy;
+
+passport.serializeUser<UserType, unknown>((user, done) => {
+    done(null, user.id)
+})
+
+passport.deserializeUser((obj, done) => {
+    // test see: https://github.com/passport/express-4.x-facebook-example/blob/master/server.js
+    done(null, obj)
+    
+    // example for mongodb
+    // User.findById(id, (err, user) => {
+    //     done(err, user);
+    // });
+})
+
+
+passport.use(new GithubStrategy({
+    clientID: config.GITHUB_CLIENT_ID ? config.GITHUB_CLIENT_ID : '',
+    clientSecret: config.GITHUB_CLIENT_SECRET ? config.GITHUB_CLIENT_SECRET : '',
+    callbackURL: 'http://localhost:3000/auth/github/callback' // url in frontend, best to use a constant here
+  },
+  function(_accessToken, _refreshToken, profile, cb) {
+    
+    let match:UserType|undefined = User.getUserByGithubId(profile.id)
+    if (!match) {
+      
+      const user:UserType = {
```

Comment:
Line 32: Explicitly giving type to `match` is redundant because `getUserByGithubId()` returns `UserType | undefined`

Line 35: Giving type to user is redundant because user has all the fields that `UserType` requires and typescript recognizes them as same type

Reply:
Removed redundant type defs